lib/cloudant.js

```diff
@@ -580,6 +590,59 @@ Cloudant.prototype.ping = function(cb) {
 };
 
 /**
+ * Replace if the model instance exists with the same id or create a
+ * new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [cb] The callback function
+ */
+Cloudant.prototype.replaceOrCreate = function replaceOrCreate(
+  model, data, options, cb) {
+  debug('Cloudant.prototype.replaceOrCreate %j %j', model, data);
+  var self = this;
+  var idName = self.idName(model);
+  var id = data[idName];
+  if (id) {
+    self.getCurrentRevision(model, id, function(err, rev) {
+      if (err) return cb(err);
+      if (rev) data._rev = rev;
+      self.create(model, data, options, function(err) {
+        if (err) return cb(err);
+        self.findById(model, data.id, {}, function(err, result) {
+          if (err) return cb(err);
+          cb(err, result, {isNewInstance: !rev});
+        });
+      });
+    });
+  } else {
+    self.create(model, data, options, function(err, id) {
+      if (err) return cb(err);
+      self.findById(model, data.id, {}, function(err, result) {
+        if (err) return cb(err);
+        cb(err, result, {isNewInstance: true});
+      });
+    });
+  }
```

Comment:
I create a bug issue to track race condition fix: https://github.com/strongloop/loopback-connector-cloudant/issues/38

Will provide more details when someone get chance to work on it.